src/lib/OpenEXR/ImfZstdCompressor.cpp

```diff
@@ -0,0 +1,110 @@
+#include <cstring>
+#include "ImfZstdCompressor.h"
+
+#include "blosc2.h"
+#include "IlmThreadPool.h"
+#include "ImfChannelList.h"
+#include "ImfMisc.h"
+namespace
+{
+class BloscInit
+{
+public:
+    static void Init () { getInstance (); }
+    BloscInit (const BloscInit&) = delete;
+    BloscInit& operator= (const BloscInit&) = delete;
+
+private:
+    BloscInit () { blosc2_init (); }
+    ~BloscInit () { blosc2_destroy (); }
+    static BloscInit& getInstance ()
+    {
+        static BloscInit instance;
+        return instance;
+    }
+};
+} // namespace
+
+OPENEXR_IMF_INTERNAL_NAMESPACE_SOURCE_ENTER
+ZstdCompressor::ZstdCompressor (
+    const Header& hdr, size_t maxScanlineSize, size_t numScanLines)
+    : Compressor (hdr)
+    , _maxScanlineSize (maxScanlineSize)
+    , _numScanLines (numScanLines)
+    , _outBuffer (nullptr, &free)
+    , _schunk (nullptr, &blosc2_schunk_free)
+{}
+
+int
+ZstdCompressor::numScanLines () const
+{
+    return _numScanLines; // Needs to be in sync with ImfCompressor::numLinesInBuffer
+}
+int
+ZstdCompressor::compress (
+    const char* inPtr, int inSize, int minY, const char*& outPtr)
+{
+    auto ret = BLOSC_compress_impl (inPtr, inSize, outPtr);
+    return ret;
+}
+int
+ZstdCompressor::uncompress (
+    const char* inPtr, int inSize, int minY, const char*& outPtr)
+{
+    auto ret = BLOSC_uncompress_impl (inPtr, inSize, outPtr);
+    return ret;
+}
+
+int
+ZstdCompressor::BLOSC_compress_impl (
+    const char* inPtr, int inSize, const char*& out)
+{
+    BloscInit::Init ();
+    blosc2_cparams cparams = BLOSC2_CPARAMS_DEFAULTS;
+
+    cparams.typesize = pixelTypeSize (PixelType::FLOAT); // Expect Float values
+    cparams.clevel =
+        header ()
+            .zstdCompressionLevel (); // 9 is about a 20% increase in compression compared to 5. Decompression speed is unchanged.
```

Comment:
don't like long comments inlined
```suggestion
    // clevel 9 is about a 20% increase in compression compared to 5.
    // Decompression speed is unchanged.
    cparams.clevel   = header ().zstdCompressionLevel ();
```